aten/src/ATen/native/cpu/Reduce.h

```diff
@@ -110,7 +110,7 @@ static inline void vectorized_outer_reduction(char** data, int64_t inner_stride,
 
 template<typename traits, typename res_t>
 static void set_result(const int index, const res_t result, const TensorIterator &iter, const int num_outputs) {
-  static_assert(std::is_same<res_t, typename traits::arg2_t>::value, "data types must match");
+  // static_assert(std::is_same<res_t, typename traits::arg2_t>::value, "data types must match");
```

Comment:
`arg2_t` is the same as `scalar_t` so this is assuming the input and output `dtypes` are the same. Is this safe to ignore in general?